### PR TITLE
dev/core#1403 - CRM/Contribute -  Change "Date Received" to "Contribution Date"

### PR DESCRIPTION
**Pull request by olayiwola-compucorp**
Overview
----------------------------------------
Development work for https://lab.civicrm.org/dev/core/-/issues/1403

This PR updates the  Contribution `receive_date` title/label from `Date Received` to `Contribution Date` so it's clear that the date is the contribution date.  Also, the receive_date description /commenthas been removed.

Before
----------------------------------------

The contribution `receive_date` has the title `Date Received` or `Received`.

After
----------------------------------------

The contribution `receive_date` title has now been changed to `Contribution Date`.

e.g.
New/Edit Contribution Screen
image

Contact Contributions
image

Contribution Searckit
image

New Membership Screen
Screenshot 2023-05-05 at 17 07 06

In summary, the changes will affect screens such as the Contribution Import/Export screen which previously used "Date Received" or "Received" as a label for the Contribution "receive_date" field.

### CODE REVIEW

**Comment by larssandergreen** [comment]
I have reviewed.
Code changes look good, except we need to [update the schema here](https://github.com/civicrm/civicrm-core/blob/master/xml/schema/Contribute/Contribution.xml) and then run GenCode.php (see the [top info box here](https://docs.civicrm.org/dev/en/latest/framework/database/schema-definition/)) to regenerate the DAO file, rather than editing it directly.
No issues on r-run.

It looks like the consensus in the linked issue was to change the label to "Contribution date" and this changes to "Date", but I think it is clear that it is a contribution date in context and "Contribution date" would be redundant, so that seems better to me.

There are also a lot more places where Received Date or just Received is used. I think it would make sense to update all of those at the same time in this PR.

Wouldn't it make sense to change the descriptions at the same time? Since we're saying this is not, in some cases, actually the receive date, it doesn't really make sense to have a description that says something about the date the contribution was received. I think different orgs are going to use date differently and we shouldn't be trying to describe specific use cases. I'm in favour of removing the description entirely, both the schema comment and where it appears on [forms](https://github.com/civicrm/civicrm-core/blob/88b8503d923526b5645a8bc141622080efa88c94/templates/CRM/Contribute/Form/Contribution.tpl#L196).

**Comment by kcristiano** [comment]
I agree @larssandergreen about the approved label change.  I think `Date` is too ambiguous and will cause confusion.    I'd have brought this up on the issue had it been simply Date.   I think that it needs more clarification.  What Date?   If Pay Later does this reflect the original Date or the date Payment was received?  How does it relate to Transaction Date?   I'd rather see this label as Contribution Date and not simply Date.

**Comment by magnolia61** [comment]
Agree!
"Contribution date" would be a welcome alternative. Just 'date' would be too general

**Comment by demeritcowboy** [comment]
jenkins test this please

**Comment by demeritcowboy** [comment]
jenkins retest this please

**Comment by demeritcowboy** [comment]
Just commenting based on the discussion this would be mergeable, but I haven't merged yet because the tests haven't run and I can't seem to get them to run (note it only says 2 checks have run - those are the style check and test site build, but not the actual unit tests).

I think maybe an admin needs to add to whitelist to get them to run?

**Comment by larssandergreen** [comment]
I had a quick look and looks good, but there is one "Received" still [here](https://github.com/civicrm/civicrm-core/blob/bcacd4a97e6f0ae43e25840f6318059777e83318/ext/legacycustomsearches/templates/CRM/Contact/Form/Search/Custom/FullText.tpl#L213) and another [here](https://github.com/civicrm/civicrm-core/blob/bcacd4a97e6f0ae43e25840f6318059777e83318/templates/CRM/Contribute/Form/PaymentInfoBlock.tpl#L17), another [here](https://github.com/civicrm/civicrm-core/blob/b68e64f956b341ee3d0ba6dd04bec83fde10b461/templates/CRM/Contribute/Form/Task/Print.tpl#L25), and [here](https://github.com/civicrm/civicrm-core/blob/b68e64f956b341ee3d0ba6dd04bec83fde10b461/templates/CRM/Contribute/Page/ContributionRecurPayments.tpl#L11), [here](https://github.com/civicrm/civicrm-core/blob/bcacd4a97e6f0ae43e25840f6318059777e83318/templates/CRM/Contribute/Page/ContributionSoft.tpl#L25), some "Received Date" [here](https://github.com/civicrm/civicrm-core/blob/bcacd4a97e6f0ae43e25840f6318059777e83318/templates/CRM/Contribute/Page/UserDashboard.tpl#L19) and [here](https://github.com/civicrm/civicrm-core/blob/bcacd4a97e6f0ae43e25840f6318059777e83318/templates/CRM/Contribute/Page/UserDashboard.tpl#L94).

There are more, but I'm not able to catalogue them all here right now. So I think perhaps best to wait to merge to make this change in full?

**Comment by larssandergreen** [comment]
Few more [here](https://github.com/civicrm/civicrm-core/blob/b68e64f956b341ee3d0ba6dd04bec83fde10b461/xml/templates/message_templates/pcp_owner_notify_text.tpl#L16) and [here](https://github.com/civicrm/civicrm-core/blob/b68e64f956b341ee3d0ba6dd04bec83fde10b461/xml/templates/message_templates/pcp_owner_notify_html.tpl#L22).

There are also a few in the [tests here](https://github.com/civicrm/civicrm-core/blob/master/tests/phpunit/CRM/Contact/Page/View/UserDashBoardTest.php) and [here](https://github.com/civicrm/civicrm-core/blob/master/tests/phpunit/CRM/Contribute/Import/Parser/data/soft_credit_extended.csv) that will need to be updated.

There are also two instances of "Received" in `civicrm_uf_field` in `civicrm_data.tpl`, which I believe will require changing and adding upgrade SQL to change this in existing installations.

I think that's it.

**Comment by seamuslee001** [comment]
Jenkins add to whitelist

**Comment by olayiwola-compucorp** [comment]
Thanks for the review @larssandergreen, The necessary "Received" or "Received Date" fields have all been changed to "Contribution Date".

Except for those in https://github.com/civicrm/civicrm-core/blob/master/sql/civicrm_generated.mysql, running the GenCode script doesn't update them.

**Comment by larssandergreen** [comment]
To update civicrm_generated.mysql, you need to run `bin/regen.sh`.

But when you update civicrm_data.tpl, you also need to update this data on existing installations (since this file is only used on install), which you do by adding SQL in `CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl` (you can see samples of this in upgrades for past versions in the same folder).

**Comment by demeritcowboy** [comment]
Note when you run regen.sh it will then fail unrelated tests because a recent membership wording change was merged that needs some tests updated first.

It's a bit messy right now.

Note you can also run regen online in your fork: see https://github.com/civicrm/civicrm-core/blob/master/.github/workflows/regen.yml

File: CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl [hunk 1]
```diff
@@ -12,3 +12,6 @@ INSERT IGNORE INTO civicrm_navigation
 VALUES
     ( {$domainID}, 'https://docs.civicrm.org/user/?src=iam', '{ts escape="sql" skip="true"}User Guide{/ts}', 'User Guide', NULL, 'AND', @adminHelplastID, '1', NULL, 1 ),
     ( {$domainID}, 'https://civicrm.org/help?src=iam',       '{ts escape="sql" skip="true"}Get Help{/ts}',   'Get Help',   NULL, 'AND', @adminHelplastID, '1', NULL, 2 );
+
+{* https://github.com/civicrm/civicrm-core/pull/26145 *}
+UPDATE civicrm_uf_field SET label = '{ts escape="sql"}Contribution Date{/ts}' WHERE field_name = 'receive_date';
```
Comment (demeritcowboy) [inline CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:17]: The test is failing because this would need to be done in php and accounting for multilingual where the field is called label_enUS, etc, like in FiveThirtyOne.php. But I think it would also be ok to leave it out so we can get this merged.

Comment (larssandergreen) [reply CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:17]:
```suggestion
{if $multilingual}
  {foreach from=$locales item=locale}
      UPDATE IGNORE civicrm_uf_field SET label_{$locale} = '{ts escape="sql"}Contribution Date{/ts}' WHERE field_name = 'receive_date';
  {/foreach}
{else}
UPDATE IGNORE civicrm_uf_field SET label = '{ts escape="sql"}Contribution Date{/ts}' WHERE field_name = 'receive_date';
{/if}
```
Sorry, I should have thought to mention that. This should do the job. It still works in SQL, @demeritcowboy?

Comment (larssandergreen) [reply CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:17]: I mean, it does work, I tested it. But there's no specific reason we can't still use SQL here, is what I meant.

Comment (olayiwola-compucorp) [reply CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:17]: @larssandergreen, thanks for the suggestion it looks good, is it okay to commit?
 or do we want to leave it as it is?

**Comment by magnolia61** [comment]
Would this also change the name of the token? from {contribution.receive_date} to {contribution.date}
I think that would be consistent, could not find that change but maybe I overlooked

File: CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl (hunk 1 above)
Comment (demeritcowboy) [reply CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:17]: It can't be left as is, and now there is also a merge conflict. Either sql or php but also it should only change it if the current label is Date Received, otherwise it will overwrite someone's preferred wording if they already changed it in the profile.

Comment (demeritcowboy) [reply CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl]: Hi @olayiwola-compucorp it's still failing tests since this is the same code as before. Can I suggest leaving this out and then we can merge the rest? You can always put up another PR after.

**Comment by demeritcowboy** [comment]
Yay.

@olayiwola-compucorp Can you make a PR to add yourself to [contributor-key.yml](https://github.com/civicrm/civicrm-core/blob/master/contributor-key.yml) for release notes credit?

**Comment by olayiwola-compucorp** [comment]
Thanks @demeritcowboy, done https://github.com/civicrm/civicrm-core/pull/26225

**Comment by jamienovick** [comment]
Great job @olayiwola-compucorp and a big thank you to everyone on this ticket who has got this over the line. One small change for civi, a giant leap for civi-kind!
